src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2593,24 +2588,36 @@ int wm_vuldet_check_agent_vulnerabilities(wm_vuldet_t *vuldet) {
     }
 
     //Check which type of scan is required
-    time_t last_full_scan = wm_vuldet_get_last_full_scan(&scan_ctx);
-    if (last_full_scan < 0) {
-        // DB error
-        result = OS_INVALID;
-        goto end;
-    }
-    else if (last_full_scan == 0) {
-        scan_ctx.scan_type = VU_BASELINE_SCAN;
-        mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_AG_BASELINE_SCAN, scan_ctx.agent_id);
-    }
-    else if (curr_time - last_full_scan >= vuldet->min_full_scan_interval &&
-            wm_vuldet_feed_changed (vuldet->updates, agents_it, last_full_scan)) {
-        scan_ctx.scan_type = VU_FULL_SCAN;
-        mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_AG_FULL_SCAN, scan_ctx.agent_id);
+    if (scan_type != VU_UNKNOWN_SCAN) {
+        scan_ctx.scan_type = scan_type;
+    } else {
+        time_t last_full_scan = wm_vuldet_get_last_full_scan(&scan_ctx);
+        if (last_full_scan < 0) {
+            // DB error
+            result = OS_INVALID;
+            goto end;
+        } else if (last_full_scan == 0) {
+            scan_ctx.scan_type = VU_BASELINE_SCAN;
+        } else if (curr_time - last_full_scan >= vuldet->min_full_scan_interval &&
+                wm_vuldet_feed_changed (vuldet->updates, agents_it, last_full_scan)) {
+            scan_ctx.scan_type = VU_FULL_SCAN;
+        } else {
+            scan_ctx.scan_type = VU_PARTIAL_SCAN;
+        }
     }
-    else {
-        scan_ctx.scan_type = VU_PARTIAL_SCAN;
-        mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_AG_PART_SCAN, scan_ctx.agent_id);
+
+    switch (scan_ctx.scan_type) {
```

Comment:
In this case, the `switch` statement isn't necessary.
We could simply write
```
mtdebug1(WM_VULNDETECTOR_LOGTAG, scan_ctx.scan_type, scan_ctx.agent_id);
```

Reply:
I implemented the changes as we discussed.

Reply:
It's great now!